test/integration/test_expid.py

```diff
@@ -260,3 +279,337 @@ def test_copy_expid_no(autosubmit: Autosubmit) -> None:
     hpc_result = describe[4].lower()
 
     assert hpc_result != new_hpc
+
+
+def create_as_exp(tmp_path):
+    _user = getuser()
+
+    return {
+        'PLATFORMS': {
+            'pytest-ps': {
+                'type': 'ps',
+                'host': '127.0.0.1',
+                'user': _user,
+                'project': 'whatever',
+                'scratch': str(tmp_path / 'scratch'),
+                'DISABLE_RECOVERY_THREADS': 'True'
+            }
+        },
+        'JOBS': {
+            'debug': {
+                'SCRIPT': 'echo "Hello world"',
+                'RUNNING': 'once'
+            },
+        }
+    }
+
+
+@pytest.mark.parametrize('test, operational, evaluation, expected, new_exp', [
+    (False, False, False, "a000", False),
+    (True, False, False, "t000", False),
+    (False, True, False, "o000", False),
+    (False, False, True, "e000", False),
+    (False, False, False, "a007", True),
+    (True, False, False, "t0ac", True),
+    (False, True, False, "o113", True),
+    (False, False, True, "e113", True),
+    (False, False, True, "", True)
+], ids=["Experiment", "Test Experiment", "Operational Experiment", "Evaluation Experiment", "New Experiment",
+        "New Test Experiment", "New Operational Experiment", "New Evaluation Experiment",
+        "New Evaluation Experiment With Empty Current"])
+def test_create_new_experiment(mocker, test, operational, evaluation, expected, new_exp):
+    db_common_mock = mocker.patch('autosubmit.experiment.experiment_common.db_common')
+    if new_exp:
+        current_experiment_id = expected
+    else:
+        current_experiment_id = "empty"
+    _build_db_mock(current_experiment_id, db_common_mock, mocker)
+    experiment_id = new_experiment(_DESCRIPTION, _VERSION, test, operational, evaluation)
+    assert expected == experiment_id
+
+
+def test_copy_experiment_new(mocker):
+    db_common_mock = mocker.patch('autosubmit.experiment.experiment_common.db_common')
+    current_experiment_id = "empty"
+    _build_db_mock(current_experiment_id, db_common_mock, mocker)
+    experiment_id = copy_experiment(current_experiment_id, _DESCRIPTION, _VERSION, False, False, True)
+    assert "" == experiment_id
+
+
+def _build_db_mock(current_experiment_id, mock_db_common, mocker):
+    mock_db_common.last_name_used = mocker.Mock(return_value=current_experiment_id)
+    mock_db_common.check_experiment_exists = mocker.Mock(return_value=False)
+
+
+def test_autosubmit_generate_config(mocker, autosubmit):
+    expid = 'ff99'
+    original_local_root_dir = BasicConfig.LOCAL_ROOT_DIR
+    read_files_mock = mocker.patch('autosubmit.autosubmit.read_files')
+
+    with tempfile.TemporaryDirectory() as temp_dir:
```

Comment:
Can be replaced by `tmp_path` or `tmpdir` fixtures (tmp_path is useful as it's easier to access later if needed, in your temp dir, tmpdir will be destroyed if I remember well)